erpnext/projects/doctype/project_task/project_task.py

```diff
@@ -7,3 +7,6 @@
 
 class ProjectTask(Document):
 	pass
+
+def on_doctype_update():
+	frappe.db.add_index("Project Task", ["task_id"])
```

Comment:
To create index based on single field, don't need to write code. It can be added based on docfield property